### PR TITLE
Updated legacy account type to Klaytn account type

### PR DESCRIPTION
**Pull request by jeongkyun-oh**
## Proposed changes

- Ethereum account type is removed and substitute them with KIaytn account type
- Ethereum account try to save storage, so it implemented `SlimAccount` type which substitute `emptyHash` and `emptyCodeHash` to `nil` value. In this PR, the `SlimAccount` concept is deleted since we have two different types of Account.
- Simply put, we do encoding/decoding `account.Account` type.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: snapshot/generate.go [hunk 1]
```diff
@@ -615,31 +616,27 @@ func (dl *diskLayer) generate(stats *generatorStats) {
...
-		if err := rlp.DecodeBytes(val, &acc); err != nil {
+		serializer := account.NewAccountSerializer()
+		if err := rlp.DecodeBytes(val, serializer); err != nil {
 			logger.Crit("Invalid account encountered during snapshot creation", "err", err)
 		}
+		acc := serializer.GetAccount()
 		// If the account is not yet in-progress, write it out
 		if accMarker == nil || !bytes.Equal(accountHash[:], accMarker) {
 			dataLen := len(val) // Approximate size, saves us a round of RLP-encoding
+			// TODO-Klaytn-Snapshot consider slim account in order to save space for emptyRootHash, emptyCodeHash
```
Comment (kjhman21) [inline snapshot/generate.go:627]: @jeongkyun-oh Do we need something more to implement here?

File: snapshot/generate_test.go [hunk 2]
```diff
@@ -118,6 +149,10 @@ func hashData(input []byte) common.Hash {
 
 // Tests that snapshot generation with existent flat state.
 func TestGenerateExistentState(t *testing.T) {
+	if true {
+		enableLog()
+	}
```
Comment (kjhman21) [inline snapshot/generate_test.go:154]: why do we need this condition?

File: snapshot/generate.go (hunk 1 above)
Comment (jeongkyun-oh) [reply snapshot/generate.go:627]: @kjhman21 The original implementation modified their account model to reduce data size. So, we could do that also but it is optional so I leave a comment. It is fine as it is, though.

File: snapshot/generate_test.go (hunk 2 above)
Comment (jeongkyun-oh) [reply snapshot/generate_test.go:154]: This is not necessary, I will remove this.

File: snapshot/generate.go (hunk 1 above)
Comment (kjhman21) [reply snapshot/generate.go:627]: @jeongkyun-oh Can we do more? I think we do not have emptyRootHash or codeHash for EOAs.

File: snapshot/generate_test.go [hunk 3]
```diff
@@ -118,6 +149,10 @@ func hashData(input []byte) common.Hash {
 
 // Tests that snapshot generation with existent flat state.
 func TestGenerateExistentState(t *testing.T) {
+	if false {
```
Comment (kjhman21) [inline snapshot/generate_test.go:152]: hmm??? false??
@jeongkyun-oh What about adding a comment here something like "replace it with true if you want to see logs"

File: snapshot/generate.go (hunk 1 above)
Comment (jeongkyun-oh) [reply snapshot/generate.go:627]: @kjhman21 How about SmartContractAccount?

File: snapshot/generate.go [hunk 4]
```diff
@@ -657,7 +654,14 @@ func (dl *diskLayer) generate(stats *generatorStats) {
 		}
 		// If the iterated account is the contract, create a further loop to
 		// verify or regenerate the contract storage.
-		if acc.Root == emptyRoot {
+		contractAcc, ok := acc.(*account.SmartContractAccount)
+		if !ok {
+			accMarker = nil
+			return nil
+		}
```
Comment (aidan-kwon) [inline snapshot/generate.go:661]: Don't we need to call `wipeKeyRange` for EOA?

**Review by kjhman21 — APPROVED** [review]

File: snapshot/generate.go (hunk 4 above)
Comment (jeongkyun-oh) [reply snapshot/generate.go]: @aidan-kwon Can you check this again?

**Review by aidan-kwon — APPROVED** [review]

**Comment by aidan-kwon** [comment]
@KimKyungup @ehnuje PTAL

File: snapshot/generate.go [hunk 5]
```diff
@@ -657,7 +645,8 @@ func (dl *diskLayer) generate(stats *generatorStats) {
 		}
 		// If the iterated account is the contract, create a further loop to
 		// verify or regenerate the contract storage.
-		if acc.Root == emptyRoot {
+		contractAcc, ok := acc.(*account.SmartContractAccount)
+		if !ok {
 			// If the root is empty, we still need to ensure that any previous snapshot
```
Comment (yoomee1313) [inline snapshot/generate.go:650]: Is the `root empty` comment's location here? Shouldn't it be moved down?

File: snapshot/generate.go [hunk 6]
```diff
@@ -615,31 +616,18 @@ func (dl *diskLayer) generate(stats *generatorStats) {
...
 		// If the account is not yet in-progress, write it out
 		if accMarker == nil || !bytes.Equal(accountHash[:], accMarker) {
 			dataLen := len(val) // Approximate size, saves us a round of RLP-encoding
 			if !write {
-				if bytes.Equal(acc.CodeHash, emptyCode[:]) {
-					dataLen -= 32
-				}
-				if acc.Root == emptyRoot {
-					dataLen -= 32
-				}
```
Comment (hqjang-pepper) [inline snapshot/generate.go:637]: Why did you remove this logic? Is it unnecessary in Klaytn?

**Review by yoomee1313 — APPROVED** [review]

File: snapshot/generate.go (hunk 6 above)
Comment (jeongkyun-oh) [reply snapshot/generate.go:637]: @hqjang-pepper This is for approximate data size. So, Etheruem uses slim account in order to save the space. It reduces `dataLen` since `nil` is stored instead of `emptyRootHash` or `emptyCodeHash`.
